frontend/src/libs/components/button/button.tsx

```diff
@@ -1,13 +1,17 @@
+import { type IconColor } from '#libs/enums/icon-color.enum.js';
 import { getValidClassNames } from '#libs/helpers/helpers.js';
-import { type IconName } from '#libs/types/types.js';
+import { type IconName, type ValueOf } from '#libs/types/types.js';
 
 import { Icon } from '../components.js';
 import styles from './styles.module.scss';
 
 type Properties = {
   label: string;
   type?: 'button' | 'submit';
-  iconName?: IconName;
+  icon?: {
+    name: IconName;
+    color?: ValueOf<typeof IconColor>;
+  };
```

Comment:
let's pass icon color and return this back

Reply:
Done